core/src/main/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilder.java

```diff
@@ -81,12 +123,15 @@ public FunctionScoreQueryBuilder(ScoreFunctionBuilder scoreFunctionBuilder) {
      * @param filter the filter that defines which documents the function_score query will be executed on.
      * @param scoreFunctionBuilder score function that is executed
      */
-    public FunctionScoreQueryBuilder add(QueryBuilder filter, ScoreFunctionBuilder scoreFunctionBuilder) {
+    public FunctionScoreQueryBuilder add(QueryBuilder<?> filter, ScoreFunctionBuilder scoreFunctionBuilder) {
```

Comment:
yep doing that, I also coupled the two lists in a single object, kind of a tuple that combines filter and function.